pkg/validator/validator.go

```diff
@@ -0,0 +1,61 @@
+package validator
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/openshift/telemeter/pkg/authorize"
+	"github.com/openshift/telemeter/pkg/metricfamily"
+	"github.com/openshift/telemeter/pkg/reader"
+)
+
+type Validator struct {
+	partitionKey string
+	labels       map[string]string
+	limitBytes   int64
+	maxAge       time.Duration
+}
+
+// New handles prometheus metrics from end clients that must be assumed to be hostile.
```

Comment:
This doc has the same issue as the one for the authorizer mentioned above. This constructor func does not `handle prometheus metrics`, as written, it only returns a `*Validator` which can do that handling. More formally, it is only the `ValidateUpload` method that handles prometheus metrics from clients.